app/models/registration.ts

```diff
@@ -84,6 +81,26 @@ export default class RegistrationModel extends NodeModel.extend() {
 
     @hasMany('institution', { inverse: 'registrations' })
     affiliatedInstitutions!: DS.PromiseManyArray<InstitutionModel> | InstitutionModel[];
+
+    registrationStateMap(): any {
+        const {
+            pendingRegistrationApproval,
+            pendingEmbargoApproval,
+            pendingEmbargoTerminationApproval,
+            pendingWithdrawal,
+            withdrawn,
+            embargoed,
+        } = this;
+
+        return {
```

Comment:
Could return the destructed object above if it doesn't drive eslint mad.

Reply:
Do you have an example of that? I don't think we can return it directly like that without doing something like

`javascript
return {
    pending: this.pending,
}
`
Was kind of avoiding that.

Reply:
Thoughts on what's uglier?

Reply:
Actually because of the multi-state thing, we need to leave this as is to provide flexibility that we need.